### DISCUSSION TITLE
How to load a local file?

### DISCUSSION THREAD

**Discussion by zhouhao27** [CoderLine/alphaTab#1234] [Q&A] [answered]
I have a `test.gp` file in the same folder as `index.html`. Is it possible to load it locally? Here is my code:

```
      // initialize alphatab
      const settings = { 
        file: "./test.gp",
        player: {
          enablePlayer: true,
          soundFont: "https://cdn.jsdelivr.net/npm/@coderline/alphatab@latest/dist/soundfont/sonivox.sf2",
          scrollElement: wrapper.querySelector('.at-viewport')
        },
      };
```

But it doesn't work. Is there a way to do so? Or load file dynamically? Thanks.

**Comment by zhouhao27** [comment]
I read the `browser-advanced` example, here is the code:

```
el.ondrop = function (e) {
        e.stopPropagation();
        e.preventDefault();
        const files = e.dataTransfer.files;
        if (files.length === 1) {
            const reader = new FileReader();
            reader.onload = function (data) {
                at.load(data.target.result, [0]);
            };
            reader.readAsArrayBuffer(files[0]);
        }
    };
```

But this one need to drag the file to browser. My intention is to read local data and pass to `FileReader`. Is this possible?

**Comment by zhouhao27** [comment]
Anyway it works if I load data in iOS and convert to ArrayBuffer and pass to JS.

**Comment by Danielku15** [comment, marked as answer]
Please try not to spam the same question across many posts next time. This is not really helpful and certainly doesn't make me answer faster. 

As you indicate and as I answered [here](https://github.com/CoderLine/alphaTab/discussions/1058#discussioncomment-6645352). It is not specific to alphaTab on how to do this in iOS. alphaTab in its web version simply needs an arraybuffer or Uint8Array to load the data. How you bring your data into the browser depends on your setup and you might be better helped in an iOS/Swift community on how to get answers how to do that efficiently and best.
